tests/examples/test_examples.py

```diff
@@ -0,0 +1,128 @@
+    caikit_dir = path.abspath(path.join(path.dirname(__file__), "../.."))
+    example_dir = path.abspath(path.join(caikit_dir, "examples", example_name))
+
+    # e2e tests may be run multiple times in parallel, for that they need an
+    # input base port number which is different for every parallel branch
+    grpc_port = int(getenv("GRPC_PORT", "30085"))
+
+    # Each example test should run on a different increment (100, 200, etc.)
+    # to the base grpc port to allow for parallel test execution with no port conflict
+    grpc_port += 100
+
+    def setUp(self):
+        # Create a temporary directory
+        self.venv_dir = tempfile.mkdtemp()
+        self.python = path.abspath(path.join(self.venv_dir, "bin", "python3"))
+        self.pip = path.abspath(path.join(self.venv_dir, "bin", "pip3"))
+
+        # Create a venv, install local version of caikit and local version of requirements
+        venv.create(env_dir=self.venv_dir, system_site_packages=False, with_pip=True)
+        try:
+            subprocess.run([self.pip, "install", "-e", self.caikit_dir], check=True)
+            subprocess.run(
+                [
+                    self.pip,
+                    "install",
+                    "-r",
+                    path.join(self.example_dir, "requirements.txt"),
+                ],
+                check=True,
+            )
+        except subprocess.CalledProcessError as cpe:
+            self.fail("Could not install dependencies: {}".format(cpe))
+
+    def tearDown(self):
+        # Remove the directory after the test
+        shutil.rmtree(self.venv_dir)
+
+    def test_server_client(self):
+        # Client and server environment variables
+        env_vars = dict(
+            RUNTIME_PORT=str(self.grpc_port),
+            RUNTIME_METRICS_PORT=str(self.grpc_port + 1),
+        )
+        # Start the server
+        with subprocess.Popen(
+            [self.python, "start_runtime.py"],
+            cwd=self.example_dir,
+            env=env_vars,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        ) as server:
+            # Check if the gRPC port is open
+            # The gRPC server start-up time has some inherent variability
+            # 30s timeout should cover most situations, while keeping the
+            # test execution time reasonable
+            if not asyncio.run(waitForPort(self.grpc_port, 30)):
+                server.terminate()
+                self.fail(
+                    "Failed to connect to the gRPC server on port {} in 30s.".format(
+                        self.grpc_port
+                    )
+                )
+
+            # Server is running, start the client
+            # Use a timeout of 10s for inference. Capture outputs to report
+            # them in case of failure.
+            try:
+                subprocess.run(
+                    [self.python, path.join(self.example_dir, "client.py")],
+                    check=True,
```

Comment:
Simply check that the client returns a 0 exit code. 
More detailed checks on the grcp server API are already covered by the `runtime/*` tests.

Reply:
We need to test all code in the example, so need to use the client and server python scripts to be tested.